src/Misc/layoutbin/AgentService.js

```diff
@@ -18,19 +18,28 @@ var listener = null;
 
 var runService = function() {
     var listenerExePath = path.join(__dirname, '../bin/Agent.Listener');
-    console.log('Starting Agent listener');
+    var noStartupAsService = process.argv[2] === "noStartupAsService";
 
     if(!stopping) {
         try {
-            listener = childProcess.spawn(listenerExePath, ['run', '--startuptype', 'service'], { env: process.env });
-            console.log('started listener process');
+            if (noStartupAsService) {
+                console.log('Starting Agent listener');
+
+                listener = childProcess.spawn(listenerExePath, ['run'], { env: process.env });
+            } else {
+                console.log('Starting Agent listener with startup type: service');
+
+                listener = childProcess.spawn(listenerExePath, ['run', '--startuptype', 'service'], { env: process.env });
+            }
+
+            console.log('Started listener process');
         
             listener.stdout.on('data', (data) => {
-                console.log(data);
+                process.stdout.write(data.toString('utf8'));
```

Comment:
data is of type Buffer, so printing its contents directly to stdout results in output like this:

```
Running Azure Pipelines agent...
Starting Agent listener
started listener process
started running service
<Buffer 53 63 61 6e 6e 69 6e 67 20 66 6f 72 20 74 6f 6f 6c 20 63 61 70 61 62 69 6c 69 74 69 65 73 2e 0a>
<Buffer 43 6f 6e 6e 65 63 74 69 6e 67 20 74 6f 20 74 68 65 20 73 65 72 76 65 72 2e 0a>
<Buffer 32 30 31 39 2d 30 33 2d 30 37 20 32 30 3a 35 35 3a 32 32 5a 3a 20 4c 69 73 74 65 6e 69 6e 67 20 66 6f 72 20 4a 6f 62 73 0a>
```